client/my-sites/themes/switching-homepage-modal.jsx

```diff
@@ -0,0 +1,152 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { Dialog } from '@automattic/components';
+import {
+	getCanonicalTheme,
+	hasActivatedTheme,
+	hasAutoLoadingHomepageFeature,
+	isActivatingTheme,
+	isThemeActive,
+	shouldShowHomepageWarning,
+	getPreInstallingThemeId,
+} from 'state/themes/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import {
+	acceptSwitchingHomepageWarning,
+	hideSwitchingHomepageWarning,
+	activate as activateTheme,
+} from 'state/themes/actions';
+
+/**
+ * Style dependencies
+ */
+import './switching-homepage-modal.scss';
+
+class SwitchingHomepageModal extends Component {
```

Comment:
Should we name this `AutoLoadingHomepageModal`? I'm hoping it would be a little less confusing if we kept the nomenclature consistent.